### PR TITLE
fix: prevent mock_ai test data from leaking into telemetry

### PR DESCRIPTION
**Pull request by svarlamov**
## Summary

- **Root cause:** `record_commit_metrics` in `post_commit.rs` calls `metrics::record()` with `CommittedValues` containing `tool_model_pairs` like `["all", "mock_ai::unknown"]`, but never sets `attrs.tool`. The existing mock_ai guard in `record()` only checks `attrs.tool`, so `Committed` events with mock_ai data pass straight through to telemetry.
- **Fix:** Early-return from `record_commit_metrics` when all tools in the breakdown are mock_ai (entire event is test data), and filter individual mock_ai entries from the per-tool parallel arrays for any mixed-tool edge case.
- Adds unit tests documenting the guard limitation and verifying the filtering behavior.

## Test plan

- [x] All existing tests pass (4554 passed, 0 failed)
- [x] New unit tests verify mock_ai guard behavior
- [x] Verify `record_commit_metrics` returns early when only mock_ai tools present
- [x] Verify mock_ai entries are filtered from `tool_model_pairs` parallel arrays

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

<a href="https://app.devin.ai/review/git-ai-project/git-ai/pull/1125" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open with Devin
  </picture>
</a>

### CODE REVIEW

File: src/authorship/post_commit.rs [hunk 1]
```diff
@@ -658,8 +669,11 @@ fn record_commit_metrics(
     let mut total_ai_deletions: Vec<u32> = vec![stats.total_ai_deletions];
     let mut time_waiting_for_ai: Vec<u64> = vec![stats.time_waiting_for_ai];
```
[the remark this answers is not in the document]
Comment (svarlamov) [reply src/authorship/post_commit.rs]: Correct observation — the aggregate "all" entry at index 0 used the raw `CommitStats` totals which include mock_ai contributions from `stats_from_authorship_log`. Fixed in 19be77b5: mock_ai `tool_model_breakdown` entries are now subtracted from the aggregates (using `saturating_sub`) before building the parallel arrays.
